test/unit/cmap/handshake/client_metadata.test.ts

```diff
@@ -0,0 +1,99 @@
+import { expect } from 'chai';
+import * as os from 'os';
+
+import { makeClientMetadata } from '../../../mongodb';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const NODE_DRIVER_VERSION = require('../../../../package.json').version;
+
+describe('makeClientMetadata()', () => {
+  context('when driverInfo.platform is provided', () => {
+    it('appends driverInfo.platform to the platform field', () => {
+      const options = {
+        driverInfo: { platform: 'myPlatform' }
```

Comment:
Do we also want tests that pass combinations of the options? I was thinking we'd test both so we can ensure there's not mixing of dependency on the properties.

Reply:
the options don't impact each other at all (ex: version -> append version, but doesn't impact name or platform). testing combinations of unrelated options doesn't add value

Reply:
They don't relate in the current implementation, would we fail a test if they became related in the future? I don't feel strongly if you think it isn't necessary.